assets/components/directDebit/directDebitActions.js

```diff
@@ -67,13 +67,20 @@ function payDirectDebitClicked(callback: Function): Function {
     }
 
     checkAccount(bankSortCode, bankAccountNumber, isTestUser, csrf)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('code1');
+        }
+        return response.json();
+      })
       .then((response) => {
         if (!response.accountValid) {
           throw new Error('code1');
         }
       }).then(() => {
         callback(undefined, bankAccountNumber, bankSortCode, accountHolderName);
```

Comment:
Does this need a separate `then`?

Reply:
I prefer to have a separated function for validation and one for data processing, to increase modularity. Currently, the validation step doesn't check a lot cases, but I will add more validation cases in a future PR.